### PR TITLE
Issue #2453 - Application controller sometimes accidentally removes duplicated/excluded resource warning condition

### LINKED ISSUE #2453: Application controller sometimes accidentally removes duplicated/excluded resource warning condition
**Describe the bug**

Application controller sometimes accidentally removes duplicated/excluded resource warning condition

**To Reproduce**

The TestExcludedResource e2e test reproduces the issues

### PR DESCRIPTION
**Pull request by alexmt**
Closes #2453

### CODE REVIEW

File: pkg/apis/application/v1alpha1/types.go [hunk 1]
```diff
@@ -1816,6 +1816,17 @@ func (status *ApplicationStatus) GetErrorConditions() []ApplicationCondition {
 	return result
 }
 
+func (status *ApplicationStatus) GetConditions(conditionTypes map[ApplicationConditionType]bool) []ApplicationCondition {
```
Comment (alexec) [inline pkg/apis/application/v1alpha1/types.go]: minor - this feels weird to me - why does a getter take a map? - why not an slice?
where is the test?

**Review by alexec — CHANGES_REQUESTED** [review]
Needs tests please!

File: pkg/apis/application/v1alpha1/types.go (hunk 1 above)
Comment (alexmt) [reply pkg/apis/application/v1alpha1/types.go]: This is because golang don't have sets :( I did not want to group give list into a map

**Comment by alexmt** [comment]
added missing tests

**Review by alexec — APPROVED** [review]

**Pull request merged**